### PR TITLE
Added a Food/Hunger API

### PR DESCRIPTION
**Pull request by Parker8283**
This adds various events and utilities to help with the manipulation of Food and the Hunger system.

This code has been lifted from AppleCore, and documentation on all this can be found here: https://github.com/squeek502/AppleCore/wiki/Using-the-AppleCore-API under the first two bullet points in the Events section, and the Accessor section.

Also, tagging @squeek502, as this is his original code. I just took it out of his mod and lifted it into forge for all to have! (with permission of course :P)

### CODE REVIEW

**Comment by Geforce132** [comment]
Cool, definitely would be useful in food/hunger-based mods.

**Comment by Parker8283** [comment]
Anything missing for this to be pulled?

File: patches/minecraft/net/minecraft/util/FoodStats.java.patch [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+@@ -16,16 +16,30 @@
+     private int field_75123_d;
+     private int field_75124_e = 20;
+     private static final String __OBFID = "CL_00001729";
++    //FORGE START
++    private EntityPlayer player;
++    private int starveTimer;
++    //FORGE END
+ 
++    public FoodStats(EntityPlayer player)
```
Comment (RainWarrior) [inline patches/minecraft/net/minecraft/util/FoodStats.java.patch]: Default constructor is gone - someone might be using it

**Comment by Parker8283** [comment]
Default constructor has been re-added and deprecated.

**Comment by cpw** [comment]
Can you verify compatibility with tterrags pr please?

**Comment by Parker8283** [comment]
The only "conflict" I see between the two PRs is the fact that we both patch the ItemFood class. Also, his is for 1.7.10, but I assume it will be pulled up into 1.8. If you merge mine first, then merge his and bump to 1.8, there should be no issues. If you merge his first and pull up into 1.8, I will have to rerun genPatches so that the ItemFood patch lines up correctly.

**Comment by Parker8283** [comment]
Anything else come up with this PR?

**Comment by Parker8283** [comment]
I borked it! Thanks Git! *grumbles*

Will remake this PR with non-borkiness and inclusion of tterrag's breeding PR probably tomorrow, when I have more time.

**Pull request closed without merging**